lib/spack/spack/util/download.py

```diff
@@ -168,12 +152,34 @@ def cookie_args(cookie) -> List[str]:
         """
         return ["-j", "-b", cookie]
 
-    @property
-    def _redirect_header_args(self) -> List[str]:
-        """Redirect headers to stderr."""
-        if sys.platform == "win32":
-            return ["-D", "CON"]
-        return ["-D", "/dev/stderr"]
+    def _get_next_header(self):
+        """Returns the next header from the stream."""
+
+        # This should work for file://
+        if self._scheme not in ("https", "http"):
+            return http.client.parse_headers(self._stream), True
+
+        finished = True
+        status_line = self._stream.readline().decode("iso-8859-1")
+        if not status_line.startswith("HTTP/"):
+            raise FetchError(f"Failed to fetch {self.url}: unexpected status line: {status_line}")
+        try:
+            status = int(status_line.split()[1])
+        except ValueError:
+            raise FetchError(
+                f"Failed to fetch {self.url}: cannot parse HTTP status code from {status_line}"
+            )
+
+        if 400 <= status < 600:
+            raise FetchError(
```

Comment:
`DetailedHTTPError` would potentially be better/more consistent.

Reply:
Done